snuba/clickhouse/columns.py

```diff
@@ -90,6 +85,11 @@ def get_nested_modifiers(
         return get_nested_modifiers(self)
 
 
+class ColumnTypeWithModifier(ABC, ColumnType):
+    def __init__(self, inner_type: ColumnType) -> None:
+        self.inner_type = inner_type
+
+
```

Comment:
This seems wrong now, We have a ColumnTypeWithModifier, but get_all_modifiers is outside of it, seems this is a leaky abstraction. I assume you did this to avoid depending on this file form errors_replacer, I guess for a circular dependency. Could you please expand on what the reason was ?

Reply:
The reason is doing `is_instance(x, ModifiedColumnType and LowCardinality in x.get_all_modifiers()` everywhere to figure out if a given column is low cardinality seems tedious. Given a column type it seems like it should be easier to find the list of modifiers. Would changing the implementation help so ColumnType doesn't reference ColumnTypeWithModifier help? I just did it as a straight copy/paste to get this working quickly for now.

Reply:
Can we then split this implementation? Let the general get_all_modifiers return [] and have a specific implementation for ColumnTypeWithModifier that provides the recursive implementation? 
That way ColumnTypeWithModufier actually has some responsibility and remove the isinstance calls.

Reply:
Yep, moved this change into https://github.com/getsentry/snuba/pull/803